### ISSUE TITLE
Is there an implementation of Seqclr?

### ISSUE THREAD

**Issue by shantzhou** [amazon-research/semimtr-text-recognition#2]

**Comment by aaberdam** [comment]
Hi,
Thanks for the interest in our work!

The major components of SeqCLR are implemented in this codebase. In particular:

- The data loader is implemented in `ImageDatasetSelfSupervised`.
- The augmentation pipeline in `get_augmentation_pipeline` with `augmentation_severity=2`.
- The projection module and the instance mapping function are implemented in `SeqCLRProj`.
- Lastly, the loss function is implemented in `SeqCLRLoss`.

The main difference of this codebase with SeqCLR is the base architecture, on which we apply SeqCLR. In the paper, we used SCATTER, and here, we employ ABI-Net. 
In addition, here, we offer a combined stage of SeqCLR with supervised CE (this can be removed in the config).

**Comment by shantzhou** [comment]
> Hi, Thanks for the interest in our work!
> 
> The major components of SeqCLR are implemented in this codebase. In particular:
> 
> * The data loader is implemented in `ImageDatasetSelfSupervised`.
> * The augmentation pipeline in `get_augmentation_pipeline` with `augmentation_severity=2`.
> * The projection module and the instance mapping function are implemented in `SeqCLRProj`.
> * Lastly, the loss function is implemented in `SeqCLRLoss`.
> 
> The main difference of this codebase with SeqCLR is the base architecture, on which we apply SeqCLR. In the paper, we used SCATTER, and here, we employ ABI-Net. In addition, here, we offer a combined stage of SeqCLR with supervised CE (this can be removed in the config).
Hi,
Thank you for your reply, I am very interested in your work, I try to use seqclr to do captcha recognition, I hope to get a powerful pre-trained model.

best wishes,
Kennen Zhou

**Comment by shantzhou** [comment]
> Hi, Thanks for the interest in our work!
> 
> The major components of SeqCLR are implemented in this codebase. In particular:
> 
> * The data loader is implemented in `ImageDatasetSelfSupervised`.
> * The augmentation pipeline in `get_augmentation_pipeline` with `augmentation_severity=2`.
> * The projection module and the instance mapping function are implemented in `SeqCLRProj`.
> * Lastly, the loss function is implemented in `SeqCLRLoss`.
> 
> The main difference of this codebase with SeqCLR is the base architecture, on which we apply SeqCLR. In the paper, we used SCATTER, and here, we employ ABI-Net. In addition, here, we offer a combined stage of SeqCLR with supervised CE (this can be removed in the config).

Thank you for your reply, I am very interested in your work, I try to use seqclr to do captcha recognition, I hope to get a powerful pre-trained model.

**Comment by aaberdam** [comment]
Sounds interesting, good luck!
